File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0404.js
//404. 左叶子之和
import TreeNode from "../TreeNode.js";

const sumOfLeftLeaves = function(root) {
  if(!root) return 0;
  let leftSum=0;
  if(root.left&&!root.left.left&&!root.left.right) leftSum=root.left.val;
  return leftSum+sumOfLeftLeaves(root.left)+sumOfLeftLeaves(root.right);
};

File: 卡码/代码随想录/Python/二叉树/02-二叉树属性/LeetCode_0404.py
#404. 左叶子之和
from typing import *
from ..TreeNode import *

class Solution:
  def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
    if not root:
      return 0
    leftSum:int=0
    if root.left and not root.left.left and not root.left.right:
      leftSum=root.left.val
    return leftSum+self.sumOfLeftLeaves(root.left)+self.sumOfLeftLeaves(root.right)
